spec/requests/restaurants_spec.rb

```diff
@@ -0,0 +1,110 @@
+require 'rails_helper'
+
+RSpec.describe 'Restaurants API', type: :request do
+  # initialize test data
+  let!(:restaurants) { create_list(:restaurant, 10) }
+  let(:restaurant_id) { restaurants.first.id }
+
+  # Test suite for GET /restaurants
+  describe 'GET /restaurants' do
+    # make HTTP get request before each example
+    before { get '/restaurants' }
+
+    it 'returns restaurants' do
+      # Note `json` is a custom helper to parse JSON responses
+      expect(json).not_to be_empty
+      expect(json.size).to eq(10)
+    end
+
+    it 'returns status code 200' do
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  # Test suite for GET /restaurants/:id
+  describe 'GET /restaurants/:id' do
+    before { get "/restaurants/#{restaurant_id}" }
+
+    context 'when the record exists' do
+      it 'returns the restaurant' do
+        expect(json).not_to be_empty
+        expect(json['id']).to eq(restaurant_id)
+      end
+
+      it 'returns status code 200' do
+        expect(response).to have_http_status(200)
+      end
+    end
+
+    context 'when the record does not exist' do
+      let(:restaurant_id) { 100 }
```

Comment:
Use a faker for the id. make it very high numbers so it won't accidentally fail on something that exists